lib/webidl2.js

```diff
@@ -12,11 +12,22 @@ import { CallbackInterface } from "./productions/callback-interface.js";
 import { autoParenter } from "./productions/helpers.js";
 import { Eof } from "./productions/token.js";
 
+/** @typedef {'callbackInterface'|'dictionary'|'interface'|'mixin'|'namespace'} ExtendableInterfaces */
+/** @typedef {Object.<'extMembers', import("./productions/container.js").AllowedMember[] >} Extension */
+/** @typedef {Object.<[ExtendableInterfaces], Extension>} Extensions */
```

Comment:
This is currently not recognized as valid type expressions.

```suggestion
/** @typedef {{ extMembers?: import("./productions/container.js").AllowedMember[]}} Extension */
/** @typedef {Partial<Record<ExtendableInterfaces, Extension>>} Extensions */
```

I know it's weird to mix TypeScript into JSDoc, but it's what we have. 🥲